src/main/java/com/esri/samples/na/closest_facility/ClosestFacilitySample.java

```diff
@@ -0,0 +1,211 @@
+  private SimpleLineSymbol routeSymbol;
+  // same spatial reference of the map
+  private SpatialReference spatialReference = SpatialReferences.getWebMercator();
+
+  @Override
+  public void start(Stage stage) throws Exception {
+
+    // pane will hold mapview to be displayed on application 
+    StackPane stackPane = new StackPane();
+    Scene scene = new Scene(stackPane);
+
+    // set title, size, and add scene to stage
+    stage.setTitle("Closest Facility Sample");
+    stage.setWidth(800);
+    stage.setHeight(700);
+    stage.setScene(scene);
+    stage.show();
+
+    try {
+
+      // create a map with streets basemap and add to view
+      ArcGISMap map = new ArcGISMap(Basemap.createStreets());
+      mapView = new MapView();
+      mapView.setMap(map);
+      // add the mapview to stack pane
+      stackPane.getChildren().addAll(mapView);
+
+      // set view to be over San Diego
+      mapView.setViewpoint(new Viewpoint(32.727, -117.1750, 40000));
+
+      graphicsOverlay = new GraphicsOverlay();
+      createFacilitiesAndGraphics();
+      // to load graphics faster, add graphics overlay to view once all graphics are in graphics overlay
+      mapView.getGraphicsOverlays().add(graphicsOverlay);
+
+      // task to find the closest route between an incident and a facility 
+      final String sanDiegoRegion =
+          "http://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/ClosestFacility";
+      task = new ClosestFacilityTask(sanDiegoRegion);
+      task.loadAsync();
+
+      // symbols that display incident(black cross) and route(blue line) to view
+      SimpleMarkerSymbol incidentSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CROSS, 0xFF000000, 20);
+      routeSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0xFF0000FF, 2.0f);
+
+      // place incident were user clicks and display route to closest facility
+      mapView.setOnMouseClicked(e -> {
+        // check that the primary mouse button was clicked
+        if (e.getButton() == MouseButton.PRIMARY && e.isStillSincePress()) {
+          // show incident to the mapview
+          Point mapPoint = mapView.screenToLocation(new Point2D(e.getX(), e.getY()));
+          incidentPoint = new Point(mapPoint.getX(), mapPoint.getY(), spatialReference);
+          Graphic graphic = new Graphic(incidentPoint, incidentSymbol);
+          graphicsOverlay.getGraphics().add(graphic);
+
+          createParametersAndSolveRoute();
+        }
+      });
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Creates points around the San Diego region that will be used to create facilities and Graphics of those facilities.
+   */
+  private void createFacilitiesAndGraphics() {
+    // List of facilities to be placed around San Diego area
+    List<Point> facilityPoints = new ArrayList<>();

```

Comment:
Instead of creating a temporary list of points, I would directly create a `List<Facility>`, then do a `forEach` on it to create the graphics:
```java
// create facilities in San Diego
facilities = Arrays.asList(
  new Facility(new Point(-1.3042129900625112E7, 3860127.9479775648, spatialReference)),
  new Facility(new Point(-1.3042193400557665E7, 3862448.873041752, spatialReference)),
  new Facility(new Point(-1.3046882875518233E7, 3862704.9896770366, spatialReference)),
  new Facility(new Point(-1.3040539754780494E7, 3862924.5938606677, spatialReference)),
  new Facility(new Point(-1.3042571225655518E7, 3858981.773018156, spatialReference)),
  new Facility(new Point(-1.3039784633928463E7, 3856692.5980474586, spatialReference)),
  new Facility(new Point(-1.3049023883956768E7, 3861993.789732541, spatialReference)),
);

// create a symbol for the facilities
String facilityUrl = "http://static.arcgis.com/images/Symbols/SafetyHealth/Hospital.png";
PictureMarkerSymbol facilitySymbol = new PictureMarkerSymbol(facilityUrl);
facilitySymbol.setHeight(30);
facilitySymbol.setWidth(30);

// for each facility, create a graphic and add it to the graphics overlay
facilities.stream()
        .map(f -> new Graphic(f.getGeometry(), facilitySymbol))
        .collect(Collectors.toCollection(() -> graphicsOverlay.getGraphics()));
```